### PR TITLE
NavBar: Rework NavBar

### PR DESCRIPTION
**Pull request by robertkowalski**
Justin & I cleaned the Navbar up and changed the styling a bit.

Big internal refactor of the codebase.

Introduces `classnames` module for easy handling of classnames.

### CODE REVIEW

**Comment by robertkowalski** [comment]
one thing i really dislike (copied from the current cloudant design) is the larger font size for the username. would prefer to remove it. comments?

screen shot 2016-12-22 at 17 19 59

File: app/addons/fauxton/navigation/__tests__/login-logout-test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      username: 'Rocko',
+
+      isLoginSectionVisible: false,
+      isLoginVisibleInsteadOfLogout: false
+    };
+
+    const navBar = mount(<NavBar {...props} />);
+
+    expect(/Login/.test(navBar.text())).toBe(false);
+    expect(/Log Out/.test(navBar.text())).toBe(false);
```
Comment (millayr) [inline app/addons/fauxton/navigation/__tests__/login-logout-test.js]: Consider using Jest's `toMatch()` API for this.  It's a little cleaner for regex in my opinion.

File: app/addons/fauxton/navigation/components/Brand.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+import React, { PropTypes } from 'react';
+
+import classNames from 'classnames';
+
+const Brand = ({toggleMenu, isMinimized}) => {
```
Comment (millayr) [inline app/addons/fauxton/navigation/components/Brand.js]: where is toggleMenu used here?

**Comment by garrensmith** [comment]
I'm closing this, since the work is being continued in #883

**Pull request closed without merging**